python/ray/_private/ray_experimental_multinode_perf.py

```diff
@@ -0,0 +1,373 @@
+# coding: utf-8
+import logging
+import os
+import sys
+import time
+
+import pytest
+import torch
+
+import ray
+import ray.cluster_utils
+import ray.experimental.channel as ray_channel
+from ray.util.scheduling_strategies import NodeAffinitySchedulingStrategy
+
+logger = logging.getLogger(__name__)
+
+# Ray limits the max gRPC size to 512 MiB. We set our max size to slightly below that to
+# be safe.
+sizes = [1, 1024, 1024**2, 1024**2 * 500]
+names = ["1 byte", "1 KiB", "1 MiB", "500 MiB"]
+num_ops = [10000, 10000, 10000, 2]
+num_iterations = 3
+num_readers = [1, 2, 4, 8]
+num_cpus = 20
+
+
+def _get_node_id(self):
+    ret = ray.get_runtime_context().get_node_id()
+    time.sleep(1)
+    return ret
+
+
+@ray.remote
+def get_node_id(self):
+    return _get_node_id(self)
+
+
+def get_nodes(remote):
+    while True:
+        if remote:
+            node_ids = ray.get(
+                [get_node_id.options(num_cpus=1).remote(None) for _ in range(num_cpus)]
+            )
+            nodes = set()
+            for node_id in node_ids:
+                nodes.add(node_id)
+            assert len(nodes) == 2
+            a_node, b_node = nodes
+            if a_node != b_node:
+                break
```

Comment:
Is this loop trying to repeat until it gets two different nodes? It looks like it would break on the `assert` right now.

Reply:
This is no longer necessary. I took it out.